File: 169. Majority Element/169. Majority Element.java
class Solution {

  public int majorityElement(int[] nums) {
    int currentElement = nums[0];
    int count = 0;

    for (int n : nums) {
      if (count <= 0) {
        currentElement = n;
      }

      if (currentElement == n) {
        count++;
      } else {
        count--;
      }
    }

    return currentElement;
  }
}

File: 169. Majority Element/169. Majority Element.py
from typing import List


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        curr_element = None
        count = 0

        for n in nums:
            if count <= 0:
                curr_element = n

            if n == curr_element:
                count += 1
            else:
                count -= 1

        return curr_element
